Initial proposal for a Vim Comparison

snowiow opened the pull request:
Hi, I started creating a small comparison guide to Vim. It's mostly build up like a cheat sheet for vim users to find out quickly things they know how to do in Vim and how they could be achieved in meow. 
I also added some things which are still empty, because those are the things I used frequently in Vim but couldn't find how they are done in meow yet. 

I would also like to keep this PR open for a bit so that other people can contribute as well.

DogLooksGood commented:
mark-word and mark-symbol are used to move to the end of word/symbol. 

jump to matched paren can be partly solved by meow-block. 

meow-search will respect the direction, that's how you search backward.


emacs has builtin repeat command. 

emacs has builtin commands for upper/lower/capitcal case.

Linerre commented on VIM_COMPARISON.org:
```diff
@@ -0,0 +1,99 @@
+
+However since many people who want to give meow a try, come from a Vim / Evil background, we want to give an overview of typical Vim keybindings and how they can be achieved in meow.
+
+* Movement
```
Would it be possible to turn the following key comparison into an org table? 

At first look, I thought the leftmost column of keys also work in meow. So it could be a bit misleading. 

Something like the below might be better:
| Vim key | Meow Command |
| --------| ------------------|
| `h`        | `meow-left`|

What do you think?

snowiow replied:
I created multiple tables now, one for each headline. It's much more readable now indeed 👍

snowiow commented:
> mark-word and mark-symbol are used to move to the end of word/symbol.

ah true, thanks for the hint. I added it now to the comparison

> jump to matched paren can be partly solved by meow-block.

cool, that looks even more powerful because you don't need to be on the exact character

> meow-search will respect the direction, that's how you search backward.

yes, that would resemble the `?` in vim, which is added. Or do you mean one can do `reverse-argument` - `meow-search`. But that doesn't work for me
 
> emacs has builtin repeat command.

But it seems not to do the same as in vim. At least it doesn't repeat my last editing commands, but only the last emacs command.
 
> emacs has builtin commands for upper/lower/capitcal case.

added some examples for this as well. Thanks! 🙂

Linerre commented on VIM_COMPARISON.org:
```diff
@@ -0,0 +1,122 @@
+| F - <symbol> | ~negative argument~ - ~meow-find~ - <symbol> | backward to last symbol                                                   |
+| t - <symbol> | ~meow-till~ - <symbol>                       | till next symbol                                                          |
+| %            | ~meow-block~                                 | jump to matching paren                                                    |
+|              |                                              | not one to one replacement can jump from anywhere inside the block to end |
```
No sure why the Vim column and Meow column are empty here. Something missing perhaps? Could you double check?

snowiow replied:
I did that on purpose because of the long description but I could put it into a single row

Linerre replied:
I see. To me it looks a bit confusing though. 

I might do sth like this:
1. Add a star `*` or some similar symbol as a superscript, indicating this description has extra explanation somewhere further in this doc. 
2.  Maybe right below this very table, put the extra information/annotation for the description.

What do you think? Feel free to disagree. I'm just thinking this might make the table format more consistent

snowiow replied:
changed it now, looked really strange in github and it takes care of wrapping by itself

Linerre replied:
Yes, it looks strange in github. @DogLooksGood and @eshrh may want to have a final quick review a little bit later. 
It looks good to me.  Thank you @snowiow ^_^

eshrh commented:
Oh oops i thought I was pushing to my own fork...

Anyway, sorry for the big diff, I filled the paragraphs and reformatted the tables. I also added some bits and rewrote some of the big paragraph in the introduction.

DogLooksGood commented:
I think we should have an entry for this file. @eshrh Do you have any idea about this, heard you want to update the document.

eshrh commented:
I'm fine with the file, i didn't expect this to be so long. Can we make a directory for documentation though?

DogLooksGood commented:
We can, the only problem is that we need to list the directory in melpa recipe, otherwise the file will not be included in the build.

DogLooksGood commented:
Let's merge this firstly.
